test/home_bloc_test.dart

```diff
@@ -0,0 +1,152 @@
+      wait: const Duration(milliseconds: 300),
+      seed: () => HomeState(
+        loading: Loading.loaded,
+        hasReachedMax: false,
+        characters: mockCharactersViewData.getRange(0, 11).toList(),
+        error: null,
+      ),
+      expect: () => [
+        HomeState(
+          loading: Loading.nextPage,
+          hasReachedMax: false,
+          characters: mockCharactersViewData.getRange(0, 11).toList(),
+          error: null,
+        ),
+        HomeState(
+          loading: Loading.loaded,
+          hasReachedMax: false,
+          characters: mockCharactersViewData.getRange(0, 22).toList(),
+          error: null,
+        ),
+      ],
+    );
+
+    blocTest<HomeBloc, HomeState>(
+      'When all heroes is downloaded and requested lust page ',
+      build: () => homeBloc,
+      act: (homeBloc) => homeBloc.add(ScrolledToEnd()),
+      wait: const Duration(milliseconds: 300),
+      seed: () => HomeState(
+        loading: Loading.loaded,
+        hasReachedMax: false,
+        characters: mockCharactersViewData.getRange(0, 22).toList(),
+        error: null,
+      ),
+      expect: () => [
+        HomeState(
+          loading: Loading.nextPage,
+          hasReachedMax: false,
+          characters: mockCharactersViewData.getRange(0, 22).toList(),
+          error: null,
+        ),
+        HomeState(
+          loading: Loading.loaded,
+          hasReachedMax: false,
+          characters: mockCharactersViewData.getRange(0, 30).toList(),
+          error: null,
+        ),
+      ],
+    );
+
+    blocTest<HomeBloc, HomeState>(
+      'When lust page is downloaded and user scrolled to end again',
+      build: () => homeBloc,
+      act: (homeBloc) => homeBloc.add(ScrolledToEnd()),
+      wait: const Duration(milliseconds: 300),
+      seed: () => HomeState(
+        loading: Loading.loaded,
+        hasReachedMax: false,
+        characters: mockCharactersViewData.getRange(0, 30).toList(),
+        error: null,
+      ),
+      expect: () => [
+        HomeState(
+          loading: Loading.nextPage,
+          hasReachedMax: false,
+          characters: mockCharactersViewData.getRange(0, 30).toList(),
+          error: null,
+        ),
+        HomeState(
+          loading: Loading.loaded,
+          hasReachedMax: true,
+          characters: mockCharactersViewData.getRange(0, 30).toList(),
+          error: null,
+        ),
+      ],
+    );
+
+    blocTest<HomeBloc, HomeState>(
+        'When repository throw DataRetrieveException',
+        build: () => homeBloc,
+        act: (homeBloc) {
+          mockRepository.noConnectionException = false;
+          mockRepository.dataRetrieveException = true;
+          homeBloc.add(ScrolledToEnd());
+        },
+        wait: const Duration(milliseconds: 300),
+        verify: (_) {
+          assert(homeBloc.state.error is DataRetrieveException);
+        });
+
+    blocTest<HomeBloc, HomeState>(
+        'When repository throw NoConnectionException',
```

Comment:
`When repository throws  NoConnectionException then state contains this exception`